lib/course_planner/csv_parser.ex

```diff
@@ -0,0 +1,54 @@
+defmodule CoursePlanner.CsvParser do
```

Comment:
why did you implement a CSV parser?
CSV parsing is much more complex than this, you should probably look for a library

Reply:
I wanted to write something simple with column validation within the parser, so in this sense this is rather interpret csv than parsing it.
but why do you say it is much more complex ? like what part of it should be different or what scenarios it does not cover ?